marlowe-runtime/src/Language/Marlowe/Runtime/Core/AddressRegistry.hs

```diff
@@ -0,0 +1,92 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Language.Marlowe.Runtime.Core.AddressRegistry
+  where
+
+import Cardano.Api.Byron (NetworkId(..))
+import Data.Foldable (asum)
+import Data.Map (Map)
+import qualified Data.Map as Map
+import Data.Set (Set)
+import qualified Data.Set as Set
+import Data.String (IsString(fromString))
+import Language.Marlowe.Runtime.ChainSync.Api (Address, ScriptHash)
+import Language.Marlowe.Runtime.Core.Api
+
+-- | The hash and address for a script.
+data ScriptAddressInfo = ScriptAddressInfo
+  { scriptAddress :: Address
+  , scriptHash :: ScriptHash
+  } deriving (Show, Eq, Ord)
```

Comment:
A given version of the Marlowe script will have a unique payment PKH, but the staking part of the address will vary. (There are important use cases where the staking address will be set to something other than `NoStakingKey`.) How are we handling this in the registry and in transaction detection?